src/AttachmentPreview/AttachmentPreview.tsx

```diff
@@ -0,0 +1,111 @@
+import * as classnames from "classnames";
+import * as React from "react";
+import * as FontAwesome from "react-fontawesome";
+
+import { Button } from "../Button/Button";
+import { FlexBox } from "../flex";
+import { AttachmentFileType, FileAttachmentIcon } from "../MessagingAttachment/MessagingAttachment";
+
+import "./AttachmentPreview.less";
+
+export interface Props {
+  attachmentID: string;
+  attachmentName: string;
+  attachmentURL: string;
+  className?: string;
+  closeButtonAriaLabel?: string;
+  downloadButtonText?: string;
+  fileType: AttachmentFileType;
+  onClickDownload: (attachmentID: string) => void;
+  onClose: () => void;
+}
+
+export const cssClass = {
+  CONTAINER: "AttachmentPreview",
+  BACKGROUND: "AttachmentPreview--Background",
+  HEADER_BAR: "AttachmentPreview--HeaderBar",
+  FILE_ICON: "AttachmentPreview--FileIcon",
+  ATTACHMENT_NAME: "AttachmentPreview--AttachmentName",
+  DOWNLOAD_CONTAINER: "AttachmentPreview--DownloadContainer",
+  DOWNLOAD_BUTTON: "AttachmentPreview--DownloadButton",
+  CLOSE_BUTTON: "AttachmentPreview--CloseButton",
+  PREVIEW_WINDOW: "AttachmentPreview--PreviewWindow",
+  IMAGE_CONTAINER: "AttachmentPreview--ImageContainer",
+  FOOTER_BAR: "AttachmentPreview--FooterBar",
+};
+
+/**
+ * A full page preview for messaging attachments.
+ * Currently only used for images, but eventually will be used for PDFs and other attachment types.
+ */
+export const AttachmentPreview: React.FC<Props> = ({
+  attachmentID,
+  attachmentName,
+  attachmentURL,
+  className,
+  closeButtonAriaLabel,
+  downloadButtonText,
+  fileType,
+  onClickDownload,
+  onClose,
+}) => {
+  const ESC = 27;
+
+  function handleKeyUp(e) {
+    console.log("\ne:", e);
+    if (e.keyCode === ESC) {
+      onClose();
+    }
+  }
+
+  React.useEffect(() => {
+    window.addEventListener("keyup", handleKeyUp);
+    return () => {
+      window.removeEventListener("keyup", handleKeyUp);
+    };
+  }, []);
```

Comment:
For my understanding, can you explain what's going on here with the add and then the remove?

Reply:
Yes, I can also add a comment explaining this. In React functional components, componentDidMount and componentWillUnmount aren't used. The way to achieve this behavior is with useEffect. If there is nothing passed into the second prop of useEffect ([]), then it will only get called when the component mounts. And the return value for the function gets called upon unmount. So essentially, this is the react hooks version of 

```
componentDidMount() {
    window.addEventListener("keyup", this.handleKeyUp);
  }

  componentWillUnmount() {
    window.removeEventListener("keyup", this.handleKeyUp);
  }
```

Reply:
Ah nice, makes sense! Very smooth.